tests/test_on_data/test_gin_ecephys.py

```diff
@@ -236,7 +236,7 @@ class TestConverter(NWBConverter):
                     member=interface_kwarg, container=converter.data_interface_objects["TestRecording"].source_data
                 )
         metadata = converter.get_metadata()
-        metadata["NWBFile"].update(session_start_time=datetime.now().astimezone())
+        metadata["NWBFile"].update(session_start_time=metadata["NWBFile"]["session_start_time"].astimezone())
```

Comment:
I think this is going to change the session_start_time for all the data in gin and not all interfaces have the session_start_time in their metadata. I don't think it needs to be changed right now.

Reply:
I implemented this change as in principle also a `session_stop_time` can be extracted now, which will be inconsistent if the `session_start_time` is later than the `session_stop_time`. But if this is not an issue for you I can revert the line. @h-mayorquin What do you prefer?

Reply:
I don't know if the  `session_stop_time` is something we would actually want to automatically extract, either - there could be separate data added to the file (like behavioral tracking) that has timestamps that go beyond when the recording was deemed to have 'stopped'.

Reply:
Actually it looks like the `session_stop_time` doesn't even exist in the schema (https://nwb-schema.readthedocs.io/en/latest/format.html?highlight=session_start_time#id147) so sorry for that confusion. And yeah, what you say about `session_start_time` is true and is always something to be aware of for multiple interfaces

Reply:
>  In fact, I am a bit surprised it does not throw an error as it is now.

They are now failing due to this reason. The CI wasn't making it past the minimal tests before the fix to external forks not having access to the secrets/environment variable

Reply:
> Regarding an inconsistent `session_start_time` (that is, smaller than `session_stop_time`) that you mention, how would this line solve or address this?


The line doesn't adress this issue, but at makes sure to use consistent start and stop times. So if you ever are going to validate for consistency, this won't raise an error.

Reply:
The metadata fields covered by NWB and mapped in the conversion tools are seen in this section of our metadata JSON schema: https://github.com/catalystneuro/nwb-conversion-tools/blob/main/src/nwb_conversion_tools/schemas/metadata_schema.json#L106-L138

Though that's not quite complete (on the to-do list to make that schema more readable as well as improve the coverage of it), there's also electrode-level metadata too: https://github.com/catalystneuro/nwb-conversion-tools/blob/main/src/nwb_conversion_tools/datainterfaces/ecephys/baserecordingextractorinterface.py#L49-L56

Reply:
Basic idea is to associate pieces of metadata as adjacently to their corresponding neurodata types as possible. If there's some metadata related to the acquisition device, it should go in one of the device (https://nwb-schema.readthedocs.io/en/latest/format.html?highlight=device#device) fields, if it's session-level metadata it can go at the NWBFile or even Subject level (https://github.com/catalystneuro/nwb-conversion-tools/blob/main/src/nwb_conversion_tools/schemas/metadata_schema.json#L11-L61), if it's electrode-level then associate it per-electrode, etc.

If a given piece of metadata doesn't make too much sense in any particular field, the description of any of those objects is usually the best place for adding free-text information.

Reply:
For `spikeglx` we added a dictionary with a bunch of extra data in the description o the device:
https://github.com/catalystneuro/nwb-conversion-tools/blob/c297c3f48be249275f214a6367cd5e79cb2e038b/src/nwb_conversion_tools/datainterfaces/ecephys/spikeglx/spikeglxdatainterface.py#L132-L161